src/compiler/transformers/module/module.ts

```diff
@@ -797,8 +809,15 @@ export function transformModule(context: TransformationContext): (x: SourceFile
             case SyntaxKind.PartiallyEmittedExpression:
                 return visitPartiallyEmittedExpression(node as PartiallyEmittedExpression, valueIsDiscarded);
             case SyntaxKind.CallExpression:
+                const needsShim = node === importsAndRequiresToShim?.[0];

```

Comment:
We should perform a length check here to avoid a deopt due to out of bound indexed access. I'd suggest using `firstOrUndefined`